### ISSUE TITLE
Allow reading genotype data in compressed archives

### ISSUE THREAD

**Issue by nevrome** [poseidon-framework/poseidon-hs#237]
Maybe this could be implemented with sth. like [pipes-zlib](https://hackage.haskell.org/package/pipes-zlib). It would allow for even smaller file sizes, which in turn would simplify and speed up a lot of our operations.

Ideally poseidon-hs should recognize `.[bed|bim|geno|snp].gz` suffixes in file names and stream the respective files accordingly when reading a package.

I suggest we play around with this here and if it's possible and feasible, then we could consider adding it to the standard.

**Comment by stschiff** [comment]
yes. Note that last time I tried `pipes-zlib` sadly suffered from this bug: https://github.com/k0001/pipes-zlib/issues/16 which was actually a bug in some other library upstream. I ended up decompressing directly from lazy bytestring (https://hackage.haskell.org/package/zlib-0.6.3.0/docs/Codec-Compression-Zlib.html) before then piping it through a suitable Pipes.Parser. So, definitely possible, but definitely also requires some playing around.

**Comment by nevrome** [comment]
This was implemented in #318 and became part of Poseidon v3.0.0.
